queue/sqs.go

```diff
@@ -70,26 +69,31 @@ func (q *sqsFIFOQueue) Put(j amboy.Job) error {
 		return errors.Errorf("cannot add %s because duplicate job already exists", name)
 	}
 
+	dedupID := j.ID()
+	curStatus := j.Status()
+	curStatus.ID = dedupID
+	j.SetStatus(curStatus)
 	j.UpdateTimeInfo(amboy.JobTimeInfo{
 		Created: time.Now(),
 	})
 	jobItem, err := registry.MakeJobInterchange(j, amboy.JSON)
+	if err != nil {
+		return errors.Wrap(err, "Error converting job in Put")
+	}
 	job, err := json.Marshal(jobItem)
-	if err != nil { // is this the right way to handle this?
-		return err
+	if err != nil {
+		return errors.Wrap(err, "Error marshalling job to JSON in Put")
 	}
-	groupID := RandomString(16)
-	dedupID := j.ID()
```

Comment:
Amazon errors if the deduplication ID has spaces (accepts alphanumeric values and punctuation). Is this okay? I could have something to change spaces in the ID to underscores or something but that seems hacky. I don't know what job IDs usually are though.

Reply:
they should be unique strings. you could just trim all whitespace characters.